src/type.js

```diff
@@ -4,64 +4,38 @@ import { core } from 'metal';
 
 const validators = {
 	any: () => true,
-	array: validateType('array'),
-	bool: validateType('boolean'),
-	func: validateType('function'),
-	number: validateType('number'),
-	object: validateType('object'),
-	string: validateType('string'),
-
-	arrayOf: (validator) => {
-		return composeValidator(
-			arrayOfValidatorFactory,
-			validator
-		);
-	},
-	instanceOf: (expectedClass) => {
-		return composeValidator(
-			instanceOfValidatorFactory,
-			expectedClass
-		);
-	},
-	objectOf: (validator) => {
-		return composeValidator(
-			objectOfValidatorFactory,
-			validator
-		);
-	},
-	oneOfType: (arrayOfTypeValidators) => {
-		return composeValidator(
-			oneOfTypeValidatorFactory,
-			arrayOfTypeValidators
-		);
-	},
-	shape: (shape) => {
-		return composeValidator(
-			shapeValidatorFactory,
-			shape
-		);
-	}
+	array: primitiveTypeValidatorFactory('array'),
```

Comment:
This is much easier to read, the only thing I'd do left is to use less separate factory functions, inlining as much as possible in the actual validator definitions.

For example, for the primitive validators you can now do something similar to what I had first proposed, having no factories, just functions that call a `validateType` function with the requested type.

As for the others, their factories are not reused, so they can be inlined directly here, like this:

```js
const validators = {
  any: () => true,
  array: function() {
    return validateType('array');
  },
  bool: function() {
    return validateType('boolean');
  },
  arrayOf: function() {
    return (value, name, context) => {
      if (!Array.isArray(value)) {
        return composeError('Expected an array.', name, context);
      } else {
        //...
      }
    };
  }
};
```

That way people don't have to go looking for the functions when reading this code, the only one they'd have to look for is `validateType`, which is easier to understand just by reading its name than `primitiveTypeValidatorFactory`.